src/browser/ui/dom/DOMPropertyOperations.js

```diff
@@ -105,6 +106,11 @@ var DOMPropertyOperations = {
         return '';
       }
       return name + '=' + quoteAttributeValueForBrowser(value);
+    } else if (tagName != null && tagName.indexOf('-') >= 0) {
+      if (value == null) {
+        return '';
+      }
+      return name + '=' + quoteAttributeValueForBrowser(value);
```

Comment:
We call quoteAttributeValueForBrowser, which calls escapeTextContentForBrowser.  Is that not good enough?

What are you worried about?  Users controlling the property name?